JsonClient/JsonClient/Program.cs

```diff
@@ -63,102 +72,106 @@ static async Task Main(string[] args)
-                Type t = obj.GetType();
-                var properties =t.GetProperties();
-                foreach(var pi in properties){
-                    if(Attribute.IsDefined(pi,typeof(IsClass))){
-                      PrintObject(pi.GetValue(obj));
-                    }else{
-                        var propertyName = pi.Name;
-                        var propertyValue = pi.GetValue(obj);
-                        Console.WriteLine($"{propertyName}: {propertyValue}\n");
+    public static TObject BuildObjectByConsole<TObject>(TObject obj)
+        {
+        Type t = obj.GetType();
+            var properties =t.GetProperties();
+            foreach(var pi in properties)
+            {
+                if(!Attribute.IsDefined(pi,typeof(SkipAttribute)))
+                {
+                    if(Attribute.IsDefined(pi,typeof(IsClassAttribute)))
+                    {
+                    var instanceComplexObj = Activator.CreateInstance(pi.PropertyType);
+                    var complexObj = BuildObjectByConsole(instanceComplexObj);
+                    pi.SetValue(obj,complexObj);
                     }
-                }
-        } 
-        public static TObject GetObject<TObject>(TObject obj){
-            
-            Type t = obj.GetType();
-                var properties =t.GetProperties();
-                foreach(var pi in properties){
-                    if(!Attribute.IsDefined(pi,typeof(Skip))){
-                        if(Attribute.IsDefined(pi,typeof(IsClass))){
-                        var instanceComplexObj = Activator.CreateInstance(pi.PropertyType);
-                        var complexObj = GetObject(instanceComplexObj);
-                        pi.SetValue(obj,complexObj);
-                        }else{
-                            var propertyName = pi.Name;
-                            Console.WriteLine($"write {propertyName}");
-                            var propertyValue = Console.ReadLine();
-                            var pit = pi.PropertyType;
-                            if (pit == typeof(int)){
-                            pi.SetValue(obj,Int32.Parse(propertyValue));
-                            }else if(pit == typeof(decimal)){
-                                pi.SetValue(obj,Decimal.Parse(propertyValue));
-                            }else{
-                                pi.SetValue(obj, propertyValue);
-                            }
-                        }
+                    else
+                    {
+                        string propertyValue = ObtainValueOfUser(pi.Name);
+                        var pit = pi.PropertyType;
+                        pi.SetValue(obj, ParseValueToTheCorrect(pit, propertyValue));
+
                     }
                 }
-            return obj;
+            }
+        return obj;
         }
-        public static string GetId(){
+        public static object ParseValueToTheCorrect(object pit, string val){

```

Comment:
Looks like `pit` is always a Type parameter, can we change the method signature to avoid casting it always?

Reply:
something like 
```if(pit ==typeof(string))
                        {
                            pi.SetValue(obj,propertyValue);
                        }
                        else
                        {
                            pi.SetValue(obj, ParseValueToTheCorrect(pit, propertyValue));
                        }```

Reply:
Something like 
```suggestion
        public static object ParseValueToTheCorrect(Type pit, string val){
        ...
        
```